azure-functions/acbs-function/src/functions/activity-update-facility-covenant.js

```diff
@@ -1,56 +1,43 @@
- * If the API request fails, it throws an error with details about the request and the error.
- * If the payload is not valid or does not contain all mandatory fields, it returns an object with the missing mandatory fields.
- * If any other error occurs, it logs the error and throws a new error.
+ * Handles the amendment of a facility covenant record in the ACBS system.
+ *
+ * This function performs the following operations:
+ * 1. Validates the input payload.
+ * 2. Submits the amendment to the ACBS system.
+ * 3. Handles the response from the ACBS system and returns the result.
  *
- * @param {Object} payload - The payload containing the facilityIdentifier and acbsFacilityCovenantInput.
- * @param {string} payload.facilityIdentifier - The identifier of the facility.
- * @param {import('../../mappings/facility/facility-covenant-amend').MappedFacilityCovenantAmendment} payload.acbsFacilityCovenantInput - The acbsFacilityCovenantInput object containing the covenant details.
- * @returns {Object} - An object containing the status, timestamps of when the request was sent and received, the data sent, and the data received from the API.
- * @throws {Error} - Throws an error if the payload is invalid, if the API request to generate the covenant ID fails, or if any other error occurs.
+ * @param {Object} payload - The input payload containing the facility identifier and ACBS facility covenant input.
+ * @param {string} payload.facilityId - The identifier of the facility to be amended.
+ * @param {Object} payload.acbsFacilityCovenantInput - The ACBS facility covenant input details.
+ * @returns {Object} - The result of the facility covenant record amendment, including status, timestamps, and data sent/received.
+ * @throws {Error} - Throws an error if the input payload is invalid or if there is an error during the amendment process.
  */
 const handler = async (payload) => {
   try {
     if (!payload) {
       throw new Error('Invalid facility covenant amendment payload');
     }
 
-    const { facilityIdentifier, acbsFacilityCovenantInput } = payload;
+    /**
+     * There are no mandatory fields for guarantee amendment payload
+     * `expirationDate` is only set for cover end date attrbiute.
+     * `targetAmount` is only set for the value attribute.
+     */
 
-    // Call update covenant API
+    const { facilityId, acbsFacilityCovenantInput } = payload;
     const submittedToACBS = getNowAsIsoString();
-    const { status, data } = await api.updateFacilityCovenant(facilityIdentifier, acbsFacilityCovenantInput);
+    const { status, data } = await api.updateFacilityCovenant(facilityId, acbsFacilityCovenantInput);
 
     if (isHttpErrorStatus(status)) {
       throw new Error(
         JSON.stringify(
           {
             name: 'ACBS Facility Covenant amend error',
-            facilityIdentifier,
```

Comment:
is there a reason we dont want to have an identifier for the error? Would this not make it harder to track any side effects (if, for instance, there is only a partial write of information)